### PR TITLE
[EG] maderequired

### PR DESCRIPTION
**Pull request by l0lawrence**
Making all properties required for languages that will generate off of 2024-01-01 version, csharp will generate off of 2018-01-01 and not be impacted

### CODE REVIEW

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Storage/Storage.tsp [hunk 1]
```diff
@@ -5,160 +5,202 @@ namespace Microsoft.EventGrid.SystemEvents {
...
 
     /** The etag of the blob at the time this event was triggered. */
-    eTag?: string;
+    @madeRequired(ServiceApiVersions.v2024_01_01)
+    eTag: string;
 
     /** The content type of the blob. This is the same as what would be returned in the Content-Type header from the blob. */
-    contentType?: string;
+    @madeRequired(ServiceApiVersions.v2024_01_01)
+    contentType: string;
```
Comment (JoshLove-msft) [inline specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Storage/Storage.tsp:25]: Content type should probably remain optional - https://jolov.azurewebsites.net/

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Storage/Storage.tsp [hunk 2]
```diff
@@ -5,160 +5,202 @@ namespace Microsoft.EventGrid.SystemEvents {
...
   model StorageBlobCreatedEventData {
     /** The name of the API/operation that triggered this event. */
-    api?: string;
+    @madeRequired(ServiceApiVersions.v2024_01_01)
+    api: string;
 
     /** A request id provided by the client of the storage API operation that triggered this event. */
-    clientRequestId?: string;
+    @madeRequired(ServiceApiVersions.v2024_01_01)
+    clientRequestId: string;
```
Comment (JoshLove-msft) [inline specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Storage/Storage.tsp:13]: I think this one is still optional from my testing.

Comment (JoshLove-msft) [reply specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Storage/Storage.tsp:13]: nevermind it is required on this event

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Storage/Storage.tsp [hunk 3]
```diff
@@ -173,69 +215,88 @@ namespace Microsoft.EventGrid.SystemEvents {
...
   model StorageBlobTierChangedEventData {
     /** The name of the API/operation that triggered this event. */
-    api?: string;
+    @madeRequired(ServiceApiVersions.v2024_01_01)
+    api: string;
 
     /** A request id provided by the client of the storage API operation that triggered this event. */
-    clientRequestId?: string;
+    @madeRequired(ServiceApiVersions.v2024_01_01)
+    clientRequestId: string;
```
Comment (JoshLove-msft) [inline specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Storage/Storage.tsp:238]: This is optional for blob tier changed.

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Storage/Storage.tsp [hunk 4]
```diff
@@ -173,69 +215,88 @@ namespace Microsoft.EventGrid.SystemEvents {
...
 
     /** The request id generated by the storage service for the storage API operation that triggered this event. */
-    requestId?: string;
+    @madeRequired(ServiceApiVersions.v2024_01_01)
+    requestId: string;
 
     /** The content type of the blob. This is the same as what would be returned in the Content-Type header from the blob. */
-    contentType?: string;
+    @madeRequired(ServiceApiVersions.v2024_01_01)
+    contentType: string;
```
Comment (JoshLove-msft) [inline specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Storage/Storage.tsp:246]: ContentType is optional for all storage events.

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.AVS/AVS.tsp [hunk 5]
```diff
@@ -59,38 +61,42 @@ model AvsScriptExecutionCancelledEventData
...
 
 /** Schema of the Data property of an EventGridEvent for Microsoft.AVS/clusters events. */
 model AvsClusterEventData {
   /** Id of the operation that caused this event. */
-  operationId?: string;
+  operationId: string;
 
   /** Hosts added to the cluster in this event, if any. */
-  addedHostNames?: string[];
+  @madeRequired(ServiceApiVersions.v2024_01_01)
```
Comment (JoshLove-msft) [inline specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.AVS/AVS.tsp:79]: Not needed for arrays either.

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Maps/Maps.tsp [hunk 6]
```diff
@@ -16,35 +18,41 @@ model MapsGeofenceResultEventData extends MapsGeofenceEventProperties {}
...
-  expiredGeofenceGeometryId?: string[];
+  @madeRequired(ServiceApiVersions.v2024_01_01)
+  expiredGeofenceGeometryId: string[];
 
   /** Lists the fence geometries that either fully contain the coordinate position or have an overlap with the searchBuffer around the fence. */
   geometries: MapsGeofenceGeometry[];
 
   /** Lists of the geometry ID of the geofence which is in invalid period relative to the user time in the request. */
-  invalidPeriodGeofenceGeometryId?: string[];
+  @madeRequired(ServiceApiVersions.v2024_01_01)
```
Comment (JoshLove-msft) [inline specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.Maps/Maps.tsp:28]: Can leave off the decorator for arrays

**Comment by raych1** [comment]
@msyyc 
APIView fails to generate with the following error.
```
10:04:52.700 cmdout 	[automation_generate.sh] ERROR:__main__:Fail to generate ApiView token file for azure-eventgrid: Command '['apistubgen', '--pkg-path', '.']' returned non-zero exit status 1.
```

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.AVS/AVS.tsp [hunk 7]
```diff
@@ -59,38 +61,38 @@ model AvsScriptExecutionCancelledEventData
...
-  addedHostNames?: string[];
+  addedHostNames: string[];
 
   /** Hosts removed from the cluster in this event, if any. */
-  removedHostNames?: string[];
+  removedHostNames: string[];
 
   /** Hosts in Maintenance mode in the cluster, if any. */
-  inMaintenanceHostNames?: string[];
+  inMaintenanceHostNames: string[];
```
Comment (JoshLove-msft) [inline specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.AVS/AVS.tsp:85]: Do we know if the key is still included in the event if there are no hosts?

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.AVS/AVS.tsp [hunk 8]
```diff
@@ -59,38 +61,38 @@ model AvsScriptExecutionCancelledEventData
...
-  operationId?: string;
+  operationId: string;
 
   /** Cmdlet referenced in the execution that caused this event. */
-  cmdletId?: string;
+  cmdletId: string;
 
   /** Stdout outputs from the execution, if any. */
-  output?: string[];
+  output: string[];
```
Comment (JoshLove-msft) [inline specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.AVS/AVS.tsp:97]: I think this one should still be optional.

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.PolicyInsights/PolicyInsights.tsp [hunk 9]
```diff
@@ -23,21 +23,21 @@ namespace Microsoft.EventGrid.SystemEvents {
...
 
     /** The compliance state of the resource with respect to the policy assignment. */
-    complianceState?: string;
+    complianceState: string;
 
     /** The subscription ID of the resource. */
-    subscriptionId?: string;
+    subscriptionId: string;
 
     /** The compliance reason code. May be empty. */
```
Comment (JoshLove-msft) [inline specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.PolicyInsights/PolicyInsights.tsp:40]: Does empty mean null here?

**Comment by msyyc** [comment]
> @msyyc APIView fails to generate with the following error.
> 
> ```
> 10:04:52.700 cmdout 	[automation_generate.sh] ERROR:__main__:Fail to generate ApiView token file for azure-eventgrid: Command '['apistubgen', '--pkg-path', '.']' returned non-zero exit status 1.
> ```

azure-eventgrid is customized a lot 

> @msyyc APIView fails to generate with the following error.
> 
> ```
> 10:04:52.700 cmdout 	[automation_generate.sh] ERROR:__main__:Fail to generate ApiView token file for azure-eventgrid: Command '['apistubgen', '--pkg-path', '.']' returned non-zero exit status 1.
> ```

Will take a look

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.AVS/AVS.tsp (hunk 7 above)
Comment (l0lawrence) [reply specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.AVS/AVS.tsp:85]: I do not know

File: specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.PolicyInsights/PolicyInsights.tsp (hunk 9 above)
Comment (l0lawrence) [reply specification/eventgrid/Azure.Messaging.EventGrid.SystemEvents/Microsoft.PolicyInsights/PolicyInsights.tsp:40]: I would think so

**Pull request closed without merging**